stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/attachment/internal/StreamAttachmentViewFactory.kt

```diff
@@ -0,0 +1,40 @@
+package io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.internal
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
+import io.getstream.chat.android.ui.common.extensions.internal.hasLink
+import io.getstream.chat.android.ui.message.list.adapter.internal.MessageListItemViewTypeMapper.isMedia
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.FileAttachmentsView
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.LinkAttachmentView
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.MediaAttachmentsGroupView
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.attachment.AttachmentViewFactory
+
+@InternalStreamChatApi
+internal class StreamAttachmentViewFactory : AttachmentViewFactory {
```

Comment:
Do we need a separate interface and internal implementation? Could we provide an open class to start from, like with `MessageListItemViewHolderFactory`? That way you can add a custom attachment type but otherwise delegate to the default implementation. That's the most basic use case, and it seems difficult with this interface / internal impl that we have so far. Or am I missing something?

Perhaps we could add a sample client implementation of `AttachmentViewFactory` to the `docs` module that implements something similar to the [tutorial's Imgur customization](https://getstream.io/tutorials/android-chat/#your-custom-viewholder), replacing the regular image attachment UI with something custom if the link is from imgur (doesn't have to be the overlay from the tutorial, can be just a red box in place of the image or something).

Reply:
Yes, good point. I'll make it as open class